src/main/java/com/redhat/devtools/lsp4ij/features/foldingRange/LSPFoldingRangeBuilder.java

```diff
@@ -177,14 +177,16 @@ public static TextRange getTextRange(
 
     private static int getStartOffset(@NotNull FoldingRange foldingRange, @NotNull Document document) {
         if (foldingRange.getStartCharacter() == null) {
-            return document.getLineEndOffset(foldingRange.getStartLine());
+            // Be defensive against language servers that return lines that are out of bounds for the document
+            return document.getLineEndOffset(Math.max(foldingRange.getStartLine(), 0));

```

Comment:
These two defensive changes may look strange, but I actually saw a situation where jdtls was returning a folding range of [1, 8] for a document with only 5 lines, so I've made both of these defensive to ensure that such an issue doesn't result in a runtime exception.